### PR TITLE
[NEW] Upload/reset avatar.

### LINKED ISSUE #752: [NEW] Update user avatar
From the profile view, allow to the user change its avatar picture. The idea is simple: have a action to open the [Storage Access Framework](https://developer.android.com/guide/topics/providers/document-provider.html) (only - obvious - for images) and call the [client.setAvatar(file: File, mimeType: String)](https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/blob/99b4ce6c2e79a7c6f7a4d2a9e3a8b2b719b94720/core/src/main/kotlin/chat/rocket/core/internal/rest/User.kt#L93)

Tip: see the existing upload file feature.

### LINKED ISSUE #753: [NEW] Reset user avatar
From the profile view, allow to the user reset its avatar. The idea is simple: have a action to reset the avatar calling the [client.resetAvatar(userId: String)](https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/blob/99b4ce6c2e79a7c6f7a4d2a9e3a8b2b719b94720/core/src/main/kotlin/chat/rocket/core/internal/rest/User.kt#L72).

### PR DESCRIPTION
**Pull request by philipbrito**
From gallery or by instantly taking a photo.

Depends on: https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/pull/201

Closes #752 
Closes #753
Closes #775

Note: This PR doesn't play with a better UI/UX when using this feature since there is a redesign in our roadmap in a very short term.

This removes the update avatar by URL feature per request

### CODE REVIEW

**Comment by rafaelks** [comment]
@filipedelimabrito There are conflicts here to be resolved.

File: app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt [hunk 1]
```diff
@@ -1,72 +1,141 @@
...
     private val strategy: CancelStrategy,
+    private val uriInteractor: UriInteractor,
+    val userHelper: UserHelper,
     serverInteractor: GetCurrentServerInteractor,
     factory: RocketChatClientFactory
 ) {
     private val serverUrl = serverInteractor.get()!!
     private val client: RocketChatClient = factory.create(serverUrl)
-    private lateinit var myselfId: String
+    private val myselfId = userHelper.user()?.id ?: ""
```
Comment (luciofm) [inline app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt:34]: Don't need to change now, but you can use `.orEmpty()` like:
`private val myselfId = userHelper.user()?.id.orEmpty()`

File: app/src/main/java/chat/rocket/android/db/DatabaseManager.kt [hunk 2]
```diff
@@ -330,7 +330,7 @@ class DatabaseManager(val context: Application,
             id = room.id,
             subscriptionId = subscription.id,
             type = room.type.toString(),
-            name = room.name ?: subscription.name ?: throw NullPointerException(),// this should be filtered on the SDK
+            name = room.name ?: subscription.name ?: throw NullPointerException(), // this should be filtered on the SDK
```
Comment (luciofm) [inline app/src/main/java/chat/rocket/android/db/DatabaseManager.kt:333]: 👍

**Review by luciofm — APPROVED** [review]

File: app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt (hunk 1 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt:34]: Got it! 
Helpful! Thank you!

**Pull request merged**